third_party/move/move-compiler/src/expansion/translate.rs

```diff
@@ -574,12 +578,38 @@ fn unique_attributes(
             | E::Attribute_::Assigned(n, _)
             | E::Attribute_::Parameterized(n, _) => *n,
         };
-        let name_ = match known_attributes::KnownAttribute::resolve(sym) {
-            None => E::AttributeName_::Unknown(sym),
+        let name_ = match KnownAttribute::resolve(sym) {
+            None => {
+                let flags = &context.env.flags();
```

Comment:
It would have been simpler to do this in extended_checks, plus fitting well to compiler v2, because extended_checks is using the same AST/env than that one. That is, in the new compiler, we can just run extended_checks after regular type checking.

Reply:
Simpler, yes.  But this was more educational.